src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1744,6 +1745,150 @@ class Simplex_tree {
+  /* 04 0a F(a) 0b F(b) 0c F(c) 0d F(d) 01 0b F(a,b) 00 02 0c F(b,c) 0d F(b,d) 01 0d F(b,c,d) 00 00 01 0d F(c,d) 00 00 */
+  void serialize(char* buffer, const std::size_t buffer_size) {
+    char* buffer_end = rec_serialize(&root_, buffer);
+    if (static_cast<std::size_t>(buffer_end - buffer) != buffer_size)
+      throw std::invalid_argument("Serialization does not match end of buffer");
+  }
+
+ private:
+  /** \brief Serialize each element of the sibling and recursively call serialization. */
+  char* rec_serialize(Siblings *sib, char* buffer) {
+    char* ptr = buffer;
+    ptr = Gudhi::simplex_tree::serialize_trivial(static_cast<Vertex_handle>(sib->members().size()), ptr);
+#ifdef DEBUG_TRACES
+    std::clog << "\n" << sib->members().size() << " : ";
+#endif  // DEBUG_TRACES
+    for (auto& map_el : sib->members()) {
+      ptr = Gudhi::simplex_tree::serialize_trivial(map_el.first, ptr); // Vertex
+      if (Options::store_filtration)
+        ptr = Gudhi::simplex_tree::serialize_trivial(map_el.second.filtration(), ptr); // Filtration
+#ifdef DEBUG_TRACES
+      std::clog << " [ " << map_el.first << " | " << map_el.second.filtration() << " ] ";
+#endif  // DEBUG_TRACES
+    }
+    for (auto& map_el : sib->members()) {
+      if (has_children(&map_el)) {
+        ptr = rec_serialize(map_el.second.children(), ptr);
+      } else {
+        ptr = Gudhi::simplex_tree::serialize_trivial(static_cast<Vertex_handle>(0), ptr);
+#ifdef DEBUG_TRACES
+        std::cout << "\n0 : ";
+#endif  // DEBUG_TRACES
+      }
+    }
+    return ptr;
+  }
+
+ public:
+  /** @private @brief Deserialize the array of char (flatten version of the tree) to initialize a Simplex tree.
+   * It is the user's responsibility to provide an 'empty' Simplex_tree, there is no guarantee otherwise.
+   * 
+   * @param[in] buffer A pointer on a buffer that contains a serialized Simplex_tree.
+   * @param[in] buffer_size The size of the buffer.
+   * 
+   * @exception std::invalid_argument In case the deserialization does not finish at the correct buffer_size.
+   * @exception std::logic_error In debug mode, if the Simplex_tree is not 'empty'.
+   * 
+   * @warning Serialize/Deserialize is not portable. It is meant to be read in a Simplex_tree with the same
+   * SimplexTreeOptions and on a computer with the same architecture.
+   * 
+   */
+  void deserialize(const char* buffer, const std::size_t buffer_size) {
+    GUDHI_CHECK(num_vertices() == 0, std::logic_error("Simplex_tree::deserialize - Simplex_tree must be empty"));
+    const char* ptr = buffer;
+    // Needs to read size before recursivity to manage new siblings for children
+    Vertex_handle members_size;
+    ptr = Gudhi::simplex_tree::deserialize_trivial(members_size, ptr);
+    ptr = rec_deserialize(&root_, members_size, ptr, 0);
+    if (static_cast<std::size_t>(ptr - buffer) != buffer_size) {
+      throw std::invalid_argument("Deserialization does not match end of buffer");
```

Comment:
I am not sure about doing a verification there. I could imagine serializing various objects consecutively to some buffer, and when deserializing, I may not know how long the piece of the buffer that encodes the simplex tree is, I only know that it ends somewhere and the next serialized object starts right after.
But let's keep it as is for now, we can revisit when we later have other uses or make this interface public.